packages/pyright-internal/src/common/realFileSystem.ts

```diff
@@ -453,17 +463,13 @@ export class WorkspaceFileWatcherProvider implements FileWatcherProvider, FileWa
 export class RealTempFile implements TempFile {
     private _tmpdir?: tmp.DirResult;
 
-    tmpdir() {
-        if (!this._tmpdir) {
-            this._tmpdir = tmp.dirSync({ prefix: 'pyright' });
-        }
-
-        return this._tmpdir.name;
+    tmpdir(): Uri {
+        return Uri.file(this._getTmpDir().name, /* isCaseSensitive */ true);

```

Comment:
There's a chicken and egg issue. To compute whether or not a file system is case sensitive, you need a tmp object.

Reply:
I suppose this could be used in a non case sensitive file system though. And then it might mess up appending things? But if you appended things, you'd probably be making import paths, so you'd want them to be case sensitive anyway.